flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -421,22 +423,19 @@ private ResultPartitionWriter createCollectingPartitionWriter(
 			@Override
 			public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
 				Buffer buffer = (Buffer) invocationOnMock.getArguments()[0];
-				Integer targetChannel = (Integer) invocationOnMock.getArguments()[1];
-				queues[targetChannel].add(new BufferOrEvent(buffer, targetChannel));
+				if (buffer.isBuffer()) {
+					Integer targetChannel = (Integer) invocationOnMock.getArguments()[1];
+					queues[targetChannel].add(new BufferOrEvent(buffer, targetChannel));
+				} else {
+					// is event:
+					AbstractEvent event = EventSerializer.fromBuffer(buffer, getClass().getClassLoader());
+					Integer targetChannel = (Integer) invocationOnMock.getArguments()[1];
```

Comment:
Was missing before, too, but we should recycle the buffer after creating the event.